src/app/home/tabs/feed/data-access/feed-state.service.ts

```diff
@@ -21,33 +22,43 @@ export class FeedStateService {
   private state = signal<FeedState>({
     locationRiddles: [],
     users: [],
+    loading: true,
   });
 
   // Selectors
   public locationRiddles = computed(() => this.state().locationRiddles);
   public users = computed(() => this.state().users);
+  public loading = computed(() => this.state().loading);
 
   // Action Sources (Subjects)
   public refresh = new Subject<void>();
 
   // Sources (Observables)
-  private locationRiddlesSource = this.locationRiddleApiService.getLocationRiddles()
+  private locationRiddlesSource =
+    this.locationRiddleApiService.getLocationRiddles();
   private usersSource = this.locationRiddleApiService.getLocationRiddles().pipe(
-    map((locationRiddles) => forkJoin(locationRiddles.map((locationRiddle) => this.profileApiService.getProfile(locationRiddle.userId || '')))),
-    switchMap((users) => users),
+    map((locationRiddles) =>
+      forkJoin(
+        locationRiddles.map((locationRiddle) =>
+          this.profileApiService.getProfile(locationRiddle.userId || '')
+        )
+      )
+    ),
+    switchMap((users) => users)
   );
   private refreshLocationRiddlesSource = this.refresh.pipe(
-    switchMap(() => this.locationRiddlesSource),
+    switchMap(() => this.locationRiddlesSource)
   );
   private refreshUsersSource = this.refresh.pipe(
-    switchMap(() => this.usersSource),
+    switchMap(() => this.usersSource)
   );
 
   constructor() {
     // Reducers
     connect(this.state)
       .with(this.locationRiddlesSource, (state, locationRiddles) => ({
         locationRiddles: locationRiddles,
+        loading: false,
```

Comment:
where do we set loading to true?

Reply:
It is initially set to true -> see initial state on line 25